src/js/utils/create-logger.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * @file create-logger.js
+ * @module create-logger
+ */
+import window from 'global/window';
+
+// This is the private tracking variable for the logging history.
+let history = [];
+
+/**
+ * Log messages to the console and history based on the type of message
+ *
+ * @private
+ * @param  {string} type
+ *         The name of the console method to use.
+ *
+ * @param  {Array} args
+ *         The arguments to be passed to the matching console method.
+ */
+const LogByTypeFactory = (name, log) => (type, level, args) => {
+  const lvl = log.levels[level];
+  const lvlRegExp = new RegExp(`^(${lvl})$`);
+
+  if (type !== 'log') {
+
+    // Add the type to the front of the message when it's not "log".
+    args.unshift(type.toUpperCase() + ':');
+  }
+
+  // Add console prefix after adding to history.
+  args.unshift(name + ':');
+
+  // Add a clone of the args at this point to history.
+  if (history) {
+    history.push([].concat(args));
+  }
+
+  // If there's no console then don't try to output messages, but they will
+  // still be stored in history.
+  if (!window.console) {
+    return;
+  }
+
+  // Was setting these once outside of this function, but containing them
+  // in the function makes it easier to test cases where console doesn't exist
+  // when the module is executed.
+  let fn = window.console[type];
+
+  if (!fn && type === 'debug') {
+    // Certain browsers don't have support for console.debug. For those, we
+    // should default to the closest comparable log.
+    fn = window.console.info || window.console.log;
+  }
+
+  // Bail out if there's no console or if this type is not allowed by the
+  // current logging level.
+  if (!fn || !lvl || !lvlRegExp.test(type)) {
```

Comment:
I don't recall which versions it is, but there are some IEs that don't allow using `apply`/`call` on `console` methods. Probably not the case in IE11, but might be worth checking.

Reply:
This code wasn't changed from what was around previously.

Reply:
Ah, then, it's probably safe. I could be showing my age again. 😄